go.mod

```diff
@@ -6,22 +6,24 @@ require (
 	github.com/apache/camel-k/pkg/apis/camel v1.1.0
 	github.com/apache/camel-k/pkg/client/camel v1.1.0
 	github.com/cloudevents/sdk-go/v2 v2.2.0
-	github.com/google/go-cmp v0.5.1
+	github.com/google/go-cmp v0.5.2
 	github.com/google/licenseclassifier v0.0.0-20200708223521-3d09a0ea2f39
 	github.com/influxdata/tdigest v0.0.1 // indirect
 	github.com/stretchr/testify v1.6.0 // indirect
 	go.opencensus.io v0.22.5-0.20200716030834-3456e1d174b2 // indirect
 	go.uber.org/zap v1.15.0
+	golang.org/x/tools v0.0.0-20200916195026-c9a70fc28ce3 // indirect
 	gopkg.in/yaml.v2 v2.3.0
 	gopkg.in/yaml.v3 v3.0.0-20200601152816-913338de1bd2 // indirect
-	k8s.io/api v0.18.7-rc.0
-	k8s.io/apimachinery v0.18.7-rc.0
+	honnef.co/go/tools v0.0.1-2020.1.5 // indirect
+	k8s.io/api v0.18.8
+	k8s.io/apimachinery v0.18.8
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
-	k8s.io/code-generator v0.18.6
+	k8s.io/code-generator v0.18.8
 	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
-	knative.dev/eventing v0.17.1-0.20200911130100-1fec6b5212c0
-	knative.dev/pkg v0.0.0-20200910143251-0761d6b47e4d
-	knative.dev/test-infra v0.0.0-20200910231400-cfba2288403d
+	knative.dev/eventing v0.17.1-0.20200923160140-40f5749cd065
+	knative.dev/pkg v0.0.0-20200922164940-4bf40ad82aab
+	knative.dev/test-infra v0.0.0-20200921012245-37f1a12adbd3
 )
 
 replace (
```

Comment:
I think the replace section should be changed as well, otherwise kube remains pinned to v0.17.6.
The Camel K API and client are already at version v0.18.x.